core/src/main/java/io/kestra/core/storages/InternalStorage.java

```diff
@@ -56,6 +58,23 @@ public InternalStorage(Logger logger, StorageContext context, StorageInterface s
         this.flowService = flowService;
     }
 
+    @Override
+    public KVStore namespaceKv() {
+        return new InternalKVStore(logger, context.getTenantId(), context.getNamespace(), storage);
+    }
+
+    @Override
+    public KVStore namespaceKv(String namespace) {
+        boolean isExternalNamespace = !namespace.equals(context.getNamespace());
+        // Checks whether the contextual namespace is allowed to access the passed namespace.
+        if (isExternalNamespace && flowService != null) {
+            flowService.checkAllowedNamespace(
+                context.getTenantId(), namespace, // requested Tenant/Namespace
+                context.getTenantId(), context.getNamespace() // from Tenant/Namespace
+            );
+        }
```

Comment:
This is already done in the function so why do it also here?

Reply:
Where ?